core/sds-aspect-model-aas-generator/src/test/java/io/openmanufacturing/sds/aspectmodel/aas/AspectModelAASGeneratorTest.java

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright (c) 2021, 2022 Robert Bosch Manufacturing Solutions GmbH
+ *
+ * See the AUTHORS file(s) distributed with this work for additional
+ * information regarding authorship.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package io.openmanufacturing.sds.aspectmodel.aas;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import io.adminshell.aas.v3.dataformat.DeserializationException;
+import io.adminshell.aas.v3.dataformat.xml.XmlDeserializer;
+import io.adminshell.aas.v3.model.AssetAdministrationShellEnvironment;
+import io.adminshell.aas.v3.model.ConceptDescription;
+import io.adminshell.aas.v3.model.DataSpecificationIEC61360;
+import io.adminshell.aas.v3.model.EmbeddedDataSpecification;
+import io.adminshell.aas.v3.model.Property;
+import io.adminshell.aas.v3.model.SubmodelElement;
+import io.adminshell.aas.v3.model.SubmodelElementCollection;
+import io.openmanufacturing.sds.aspectmetamodel.KnownVersion;
+import io.openmanufacturing.sds.aspectmodel.resolver.services.VersionedModel;
+import io.openmanufacturing.sds.metamodel.Aspect;
+import io.openmanufacturing.sds.metamodel.loader.AspectModelLoader;
+import io.openmanufacturing.sds.test.TestAspect;
+import io.openmanufacturing.sds.test.TestResources;
+
+class AspectModelAASGeneratorTest {
+
+  AspectModelAASGenerator generator = new AspectModelAASGenerator();
+
+  @Test
+  void testGenerateAasxFromBammAspectWithListAndAdditionalProperty()
+      throws IOException, DeserializationException {
+    final AssetAdministrationShellEnvironment env =
+        getAssetAdministrationShellFromAspect(TestAspect.ASPECT_WITH_LIST_AND_ADDITIONAL_PROPERTY);
+    assertEquals(3, env.getConceptDescriptions().size());
+    assertEquals(1, env.getSubmodels().size());
+    assertEquals(2, env.getSubmodels().get(0).getSubmodelElements().size());
+
+    final Set<String> semanticIds =
```

Comment:
Why not `Set.of(...)` instead of `Stream.of(...).collect(Collectors.toCollection(HashSet::new))`?
(Also in the other similar code parts in this file)

Reply:
thanks for the hint. Looks much cleaner.